File: Lab 2/ProblemA2.py
import sys 
data_var = sys.stdin.read().strip().split()
n = int(data_var[0])
s = int(data_var[1])
arr = []
i = 2
while i < n + 2:
    arr.append(int(data_var[i]))
    i += 1
l = 0
r = n - 1
sum = 0
flag = False
while l < r:
    sum = arr[l] + arr[r]
    if sum == s:
        sys.stdout.write(str(l+1)+ " " + str(r+1)+ "\n")
        flag = True 
        break
    elif sum> s: 
        r = r -1
    else:
        l = l + 1
if not flag:
    sys.stdout.write("-1\n")            

File: Lab 2/Java/ProblemA2.java
import java.util.Scanner;

public class ProblemA2 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        if (!sc.hasNextInt())
            return;
        int n = sc.nextInt();
        int s = sc.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        int l = 0;
        int r = n - 1;
        boolean flag = false;
        while (l < r) {
            int sum = arr[l] + arr[r];
            if (sum == s) {
                System.out.println((l + 1) + " " + (r + 1));
                flag = true;
                break;
            } else if (sum > s) {
                r--;
            } else {
                l++;
            }
        }
        if (!flag) {
            System.out.println("-1");
        }
    }
}
